mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -626,17 +619,11 @@ pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
 }
 
 fn seednodes(ctx: &MmArc) -> P2PResult<Vec<RelayAddress>> {
-    if ctx.conf["seednodes"].is_null() {
-        if ctx.p2p_in_memory() {
-            // If the network is in memory, there is no need to use default seednodes.
-            return Ok(Vec::new());
+    json::from_value(ctx.conf.get("seednodes").unwrap_or(&json!([])).clone()).map_to_mm(|e| {
+        P2PInitError::ErrorDeserializingConfig {
+            field: "seednodes".to_owned(),
+            error: e.to_string(),
```

Comment:
difficult to read because code nests multiple operations in a single expression

I suggest separating the value retrieval into a variable to improve readability
```rust
fn seednodes(ctx: &MmArc) -> P2PResult<Vec<RelayAddress>> {
    let seednodes_val = ctx.conf.get("seednodes").unwrap_or(&json!([]));

    json::from_value(seednodes_val.clone()).map_to_mm(|e| P2PInitError::ErrorDeserializingConfig {
        field: "seednodes".to_owned(),
        error: e.to_string(),
    })
}
```

Reply:
Fine